clang/lib/Sema/TreeTransform.h

```diff
@@ -7732,7 +7732,11 @@ TreeTransform<Derived>::TransformIfStmt(IfStmt *S) {
     if (Then.isInvalid())
       return StmtError();
   } else {
-    Then = new (getSema().Context) NullStmt(S->getThen()->getBeginLoc());
+    // Discarded branch is replaced with empty CompoundStmt so we can keep
+    // proper source location for start and end of original branch, so
+    // subsequent transformations like CoverageMapping work properly
```

Comment:
In the test, you mention that we're generating coverage for the discarded regions. Is the reason that we're generating a `CompoundStmt` here? Could this use a `FIXME`?

Reply:
IMHO There is nothing to fix later. This will stay, it replaces former using of NullStmt in <then> branch and NULL in <else> branch.